### DISCUSSION TITLE
Popper element is closing when hovered on target element's children

### DISCUSSION THREAD

**Discussion by rokin04** [floating-ui/floating-ui#1225] [Q&A]
Hi Team,

I have attached my popper element to a div and this div contains children.
I want my popper to remain open when I hover on the target element's children.
How can I achieve this ? without giving target reference to all the children.

**Comment by harshagarwalnyu** [comment]
This is caused by the `hover` interaction treating `mouseleave` on the reference element as "close" — but when you move from the reference element into a child, the browser fires `mouseleave` on the reference even though the cursor is still semantically "inside" the component.

**Fix with `@floating-ui/react` — use `safePolygon()`:**

```jsx
import {
  useFloating,
  useHover,
  useInteractions,
  safePolygon,
} from '@floating-ui/react';

const { refs, floatingStyles, context } = useFloating({ open, onOpenChange });

const hover = useHover(context, {
  handleClose: safePolygon(),  // <-- this is the fix
});

const { getReferenceProps, getFloatingProps } = useInteractions([hover]);
```

`safePolygon()` draws a triangle between the cursor's current position, the reference element, and the floating element. As long as the cursor stays within that triangle (even over child elements), the floating element stays open.

---

**If using the older `@popperjs/core` / `react-popper` directly:**

Add `mouseenter` / `mouseleave` listeners on the reference's container div instead of the reference element itself, so children inherit the hover zone:

```jsx
<div
  onMouseEnter={() => setOpen(true)}
  onMouseLeave={() => setOpen(false)}
>
  {/* reference + all children */}
  <button ref={setReference}>Trigger</button>
  <ChildA />
  <ChildB />
</div>
```

This avoids the problem entirely by making the hover zone the parent container.

**Recommended:** migrate to `@floating-ui/react` — `safePolygon` is the correct architectural solution and handles all edge cases (diagonal movement to floating, cursor leaving from side, etc.).
